## Countries’ Corruption Indexes Analysis

## I. Economic Theory
<br/>

* What is Market capitalization and Gross Domestic Product (GDP)?
* Market capitalization reflects the value of a company
* Market cap to GDP ratio
* Corruption Perceptions Index (CPI)

In [2]:
# packages intended to use
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt

In [12]:
# read csv file using pandas package
gdp_data_uncleaned = pd.read_csv('/Users/rache/Downloads/files/gdp_data.csv')
gdp_data_uncleaned.set_index('Country Name', inplace=True)
gdp_data_uncleaned = gdp_data_uncleaned.iloc[:, 3:]
gdp_data_uncleaned.head()

1960         1961         1962         1963         1964  \
Country Name                                                                    
Aruba                 NaN          NaN          NaN          NaN          NaN   
Afghanistan   537777811.1  548888895.6  546666677.8  751111191.1  800000044.4   
Angola                NaN          NaN          NaN          NaN          NaN   
Albania               NaN          NaN          NaN          NaN          NaN   
Andorra               NaN          NaN          NaN          NaN          NaN   

                      1965          1966          1967          1968  \
Country Name                                                           
Aruba                  NaN           NaN           NaN           NaN   
Afghanistan   1.006667e+09  1.400000e+09  1.673333e+09  1.373333e+09   
Angola                 NaN           NaN           NaN           NaN   
Albania                NaN           NaN           NaN           NaN   
Andorra                NaN           NaN           NaN           NaN   

                      1969  ...          2011          2012          2013  \
Country Name                ...                                             
Aruba                  NaN  ...  2.549721e+09  2.534637e+09  2.701676e+09   
Afghanistan   1.408889e+09  ...  1.780429e+10  2.000160e+10  2.056107e+10   
Angola                 NaN  ...  1.117900e+11  1.280530e+11  1.367100e+11   
Albania                NaN  ...  1.289077e+10  1.231983e+10  1.277622e+10   
Andorra                NaN  ...  3.629204e+09  3.188809e+09  3.193704e+09   

                      2014          2015          2016          2017  \
Country Name                                                           
Aruba         2.765363e+09  2.919553e+09  2.965922e+09  3.056425e+09   
Afghanistan   2.048489e+10  1.990711e+10  1.801775e+10  1.886995e+10   
Angola        1.457120e+11  1.161940e+11  1.011240e+11  1.221240e+11   
Albania       1.322814e+10  1.138685e+10  1.186120e+10  1.301969e+10   
Andorra       3.271808e+09  2.789870e+09  2.896679e+09  3.000181e+09   

                      2018          2019  2020  
Country Name                                    
Aruba                  NaN           NaN   NaN  
Afghanistan   1.835388e+10  1.929110e+10   NaN  
Angola        1.013530e+11  8.881570e+10   NaN  
Albania       1.514702e+10  1.527918e+10   NaN  
Andorra       3.218316e+09  3.154058e+09   NaN  

[5 rows x 61 columns]

In [13]:
# read csv file using pandas package
market_cap_data_uncleaned = pd.read_csv('/Users/rache/Downloads/files/market_cap_usd.csv')
market_cap_data_uncleaned.set_index('Country Name', inplace=True)
market_cap_data_uncleaned = market_cap_data_uncleaned.iloc[:, 3:]
market_cap_data_uncleaned.sort_values('2019', ascending=False)

1960  1961  1962  1963  1964  1965  1966  1967  1968  \
Country Name                                                                
East Asia & Pacific   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
IDA & IBRD total      NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
IBRD only             NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Low & middle income   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Middle income         NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
...                   ...   ...   ...   ...   ...   ...   ...   ...   ...   
Samoa                 NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Kosovo                NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Yemen, Rep.           NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Zambia                NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Zimbabwe              NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   

                     1969  ...          2011          2012          2013  \
Country Name               ...                                             
East Asia & Pacific   NaN  ...  1.377050e+13  1.573810e+13  1.736860e+13   
IDA & IBRD total      NaN  ...  1.036160e+13  1.161470e+13  1.154820e+13   
IBRD only             NaN  ...  1.021320e+13  1.151490e+13  1.141030e+13   
Low & middle income   NaN  ...  9.900700e+12  1.110710e+13  1.103880e+13   
Middle income         NaN  ...  9.900700e+12  1.110710e+13  1.103880e+13   
...                   ...  ...           ...           ...           ...   
Samoa                 NaN  ...           NaN           NaN           NaN   
Kosovo                NaN  ...           NaN           NaN           NaN   
Yemen, Rep.           NaN  ...           NaN           NaN           NaN   
Zambia                NaN  ...  3.184470e+09           NaN           NaN   
Zimbabwe              NaN  ...           NaN           NaN           NaN   

                             2014          2015          2016          2017  \
Country Name                                                                  
East Asia & Pacific  1.950630e+13  2.161100e+13  2.117260e+13  2.657260e+13   
IDA & IBRD total     1.325330e+13  1.427330e+13  1.441970e+13  1.777650e+13   
IBRD only            1.311710e+13  1.415720e+13  1.422760e+13  1.765310e+13   
Low & middle income  1.281180e+13  1.390960e+13  1.403060e+13  1.723680e+13   
Middle income        1.281180e+13  1.390960e+13  1.403060e+13  1.723680e+13   
...                           ...           ...           ...           ...   
Samoa                         NaN           NaN           NaN           NaN   
Kosovo                        NaN           NaN           NaN           NaN   
Yemen, Rep.                   NaN           NaN           NaN           NaN   
Zambia                        NaN           NaN           NaN           NaN   
Zimbabwe                      NaN           NaN           NaN           NaN   

                             2018          2019  2020  
Country Name                                           
East Asia & Pacific  2.171880e+13  2.382020e+13   NaN  
IDA & IBRD total     1.423380e+13  1.677690e+13   NaN  
IBRD only            1.412490e+13  1.670470e+13   NaN  
Low & middle income  1.378040e+13  1.635010e+13   NaN  
Middle income        1.378040e+13  1.634690e+13   NaN  
...                           ...           ...   ...  
Samoa                         NaN           NaN   NaN  
Kosovo                        NaN           NaN   NaN  
Yemen, Rep.                   NaN           NaN   NaN  
Zambia                        NaN           NaN   NaN  
Zimbabwe                      NaN           NaN   NaN  

[264 rows x 61 columns]

## II. Data and Summary Statistics


In [14]:
market_cap_to_gdp = market_cap_data_uncleaned.div(gdp_data_uncleaned)
market_cap_to_gdp.sort_values('2018', ascending=False, inplace=True)
market_cap_to_gdp.head(20)
market_cap_to_gdp.describe

<bound method NDFrame.describe of                       1960  1961  1962  1963  1964  1965  1966  1967  1968  \
Country Name                                                                 
Hong Kong SAR, China   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
South Africa           NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Switzerland            NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Singapore              NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
United States          NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
...                    ...   ...   ...   ...   ...   ...   ...   ...   ...   
Samoa                  NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Kosovo                 NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Yemen, Rep.            NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Zambia                 NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Zimbabwe               NaN   N

In [21]:
df = pd.read_csv('/Users/rache/Downloads/files/index.csv')
df.head(50)

CPI Rank                           Country Country Code  \
0          1                       New Zealand          NZL   
1          1                           Denmark          DNK   
2          3                           Finland          FIN   
3          4                            Sweden          SWE   
4          5                       Switzerland          CHE   
5          6                            Norway          NOR   
6          7                         Singapore          SGP   
7          8                       Netherlands          NLD   
8          9                            Canada          CAN   
9         10                           Germany          DEU   
10        10                        Luxembourg          LUX   
11        10                    United Kingdom          GBR   
12        13                         Australia          AUS   
13        14                           Iceland          ISL   
14        15                         Hong Kong          HKG   
15        15                           Belgium          BEL   
16        17                           Austria          AUT   
17        18      The United States of America          USA   
18        19                           Ireland          IRL   
19        20                             Japan          JPN   
20        21                           Uruguay          URY   
21        22                           Estonia          EST   
22        23                            France          FRA   
23        24                           Bahamas          BHS   
24        24                             Chile          CHL   
25        24              United Arab Emirates          ARE   
26        27                            Bhutan          BTN   
27        28                            Israel          ISR   
28        29                            Poland          POL   
29        29                          Portugal          PRT   
30        31                          Barbados          BRB   
31        31                            Taiwan          TWN   
32        31                             Qatar          QAT   
33        31                          Slovenia          SVN   
34        35                       Saint Lucia          LCA   
35        35  Saint Vincent and The Grenadines          VCT   
36        35                          Botswana          BWA   
37        38                          Dominica          DMA   
38        38                        Cape Verde          CPV   
39        38                         Lithuania          LTU   
40        41                        Costa Rica          CRI   
41        41                            Brunei          BRN   
42        41                             Spain          ESP   
43        44                           Georgia          GEO   
44        44                            Latvia          LVA   
45        46                           Grenada          GRD   
46        47                            Cyprus          CYP   
47        47                    Czech Republic          CZE   
48        47                             Malta          MLT   
49        50                         Mauritius          MUS   

                          Region  Corruption Perceptions Index (CPI)  \
0                   Asia Pacific                                  90   
1        Europe and Central Asia                                  90   
2        Europe and Central Asia                                  89   
3        Europe and Central Asia                                  88   
4        Europe and Central Asia                                  86   
5        Europe and Central Asia                                  85   
6                   Asia Pacific                                  84   
7        Europe and Central Asia                                  83   
8                       Americas                                  82   
9        Europe and Central Asia                                  81  

In [23]:
# list of column names
market_cap_to_gdp.columns

Index(['1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020'],
      dtype='object')

## III. Variables Descriptions
<br/>

* Corruption Perceptions Index (CPI)
* Global Insight Country Risk Ratings
* IMD World Competitiveness Yearbook
* World Justice Project Rule of Law Index
* Economist Intelligence Unit Country Ratings

In [24]:
market_cap_to_gdp.describe

<bound method NDFrame.describe of                       1960  1961  1962  1963  1964  1965  1966  1967  1968  \
Country Name                                                                 
Hong Kong SAR, China   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
South Africa           NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Switzerland            NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Singapore              NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
United States          NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
...                    ...   ...   ...   ...   ...   ...   ...   ...   ...   
Samoa                  NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Kosovo                 NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Yemen, Rep.            NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Zambia                 NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
Zimbabwe               NaN   N

### Equation

Market cap to GDP ratio = a + bXi + cXi + dXi + eXi + fXi + ϵ

![image.png](attachment:e29d2526-338f-45b5-a4c8-d4a2f29edbd6.png)

Where:
* Corruption Perceptions Index (CPI) = bX1
* Global Insight Country Risk Ratings = cX2
* IMD World Competitiveness Yearbook = dX3
* World Justice Project Rule of Law Index = eX4
* Economist Intelligence Unit Country Ratings = fX5
* Residual (error) = ϵ 

In [25]:
#Original
from pandas import DataFrame
import statsmodels.api as sm

corruption_index_data = df.fillna(method='ffill')
corruption_index_data.rename(columns={'Country': 'Country Name'}, inplace=True)

merged_data = pd.merge(corruption_index_data, market_cap_to_gdp, how='inner', on=['Country Name'])
merged_data.set_index('Country Name', inplace=True)
merged_data.dropna(subset = ['Corruption Perceptions Index (CPI)','Global Insight Country Risk Ratings','2017'],inplace=True)

x = merged_data[['Corruption Perceptions Index (CPI)','Global Insight Country Risk Ratings']]
x = sm.add_constant(x) #adding a constant

y = market_cap_to_gdp
y = merged_data[['2017']]

model = sm.OLS(y, x).fit()
predictions = model.predict(x)

print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                   2017   R-squared:                       0.163
Model:                            OLS   Adj. R-squared:                  0.133
Method:                 Least Squares   F-statistic:                     5.452
Date:                Thu, 23 Mar 2023   Prob (F-statistic):            0.00687
Time:                        22:42:35   Log-Likelihood:                -49.782
No. Observations:                  59   AIC:                             105.6
Df Residuals:                      56   BIC:                             111.8
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                                          coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------------------
co

In [27]:
!pip install linearmodels

In [28]:
#COPY using another model 
import pandas as pd
import numpy as np
from pandas import DataFrame
import matplotlib.pyplot as plt
import statsmodels.api as sm
from linearmodels import PanelOLS


corruption_index_data = df.fillna(method='ffill')
corruption_index_data.rename(columns={'Country': 'Country Name'}, inplace=True)

merged_data = pd.merge(corruption_index_data, market_cap_to_gdp, how='inner', on=['Country Name'])
merged_data.set_index('Country Name', inplace=True)
merged_data.dropna(subset = ['Corruption Perceptions Index (CPI)','Global Insight Country Risk Ratings','2017'],inplace=True)

# exog_vars = ['Corruption Perceptions Index (CPI)', 'Global Insight Country Risk Ratings']
# exog = sm.add_constant(merged_data[exog_vars])

# merged2 = merged_data.rename(columns = {'2017': 'Year'}, inplace = False)


# mod = PanelOLS(y = merged2['Year'], x = merged2[['Corruption Perceptions Index (CPI)', 'Global Insight Country Risk Ratings']], time_effects=True)

# # mod = PooledOLS(merged2.Year, exog)
# pooled_res = mod.fit()
# print(pooled_res)

x = merged_data[['Corruption Perceptions Index (CPI)','Global Insight Country Risk Ratings']]
x = sm.add_constant(x) #adding a constant

y = market_cap_to_gdp
y = merged_data[['2017']]

model = sm.OLS(y, x).fit()
predictions = model.predict(x)

print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                   2017   R-squared:                       0.163
Model:                            OLS   Adj. R-squared:                  0.133
Method:                 Least Squares   F-statistic:                     5.452
Date:                Thu, 23 Mar 2023   Prob (F-statistic):            0.00687
Time:                        22:43:22   Log-Likelihood:                -49.782
No. Observations:                  59   AIC:                             105.6
Df Residuals:                      56   BIC:                             111.8
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                                          coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------------------
co

In [29]:
#Copy for adding more variables
from pandas import DataFrame
import statsmodels.api as sm

corruption_index_data = df.fillna(method='ffill')
corruption_index_data.rename(columns={'Country': 'Country Name'}, inplace=True)

merged_data = pd.merge(corruption_index_data, market_cap_to_gdp, how='inner', on=['Country Name'])
merged_data.set_index('Country Name', inplace=True)
merged_data.dropna(subset = ['Corruption Perceptions Index (CPI)','Global Insight Country Risk Ratings','IMD World Competitiveness Yearbook','World Justice Project Rule of Law Index','Economist Intelligence Unit Country Ratings','2017'],inplace=True)

x = merged_data[['Corruption Perceptions Index (CPI)','Global Insight Country Risk Ratings','IMD World Competitiveness Yearbook','World Justice Project Rule of Law Index','Economist Intelligence Unit Country Ratings']]
x = sm.add_constant(x) #adding a constant

y = market_cap_to_gdp
y = merged_data[['2017']]

model = sm.OLS(y, x).fit()
predictions = model.predict(x)

print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                   2017   R-squared:                       0.238
Model:                            OLS   Adj. R-squared:                  0.166
Method:                 Least Squares   F-statistic:                     3.309
Date:                Thu, 23 Mar 2023   Prob (F-statistic):             0.0113
Time:                        22:43:32   Log-Likelihood:                -47.015
No. Observations:                  59   AIC:                             106.0
Df Residuals:                      53   BIC:                             118.5
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                                                  coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------------

## IV. Regression Model Results
<br/>

* We expect an average of 0.0152 increase in market cap to GDP ratio if other independent variables is constant.
* Global Insight Country Ratings, World Competitive Yearbook, and EIU ratings have a positive effects on Market cap to GDP ratio
* CPI and Rule of Law Index both have a negative effect on Market cap to GDP ratio

## V. Conclusion 

R2 from the model was low (0.238) which means the independent variables are not explaining much in the change of Market cap to GDP ratio
